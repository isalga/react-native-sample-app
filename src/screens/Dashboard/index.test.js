import React from 'react'
import ReactDOM from 'react-dom'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Dashboard } from './index'

Enzyme.configure({ adapter: new Adapter() })
jest.mock('react-native-autocomplete-input', () => () => null)

const city = {
  id: '1',
  displayName: 'Abu Dhabi',
  name: 'Abu_Dhabi',
  country: 'united_arab_Emirates'
}

const navigation = {
  setParams: () => {},
  state: { params: {} }
}

describe('Dashboard screen component', () => {

  it('should render correctly', () => {
    const wrapper = Enzyme.shallow(
      <Dashboard
        addedCities={[city]}
        navigation={navigation}/>)
    expect(wrapper).toMatchSnapshot()
    wrapper.instance().toggleShowCityModal()
    expect(wrapper.update()).toMatchSnapshot()
  })
})
