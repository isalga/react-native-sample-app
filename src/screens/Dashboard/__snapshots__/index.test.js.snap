// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard screen component should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "ghostwhite",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "country": "united_arab_Emirates",
          "displayName": "Abu Dhabi",
          "id": "1",
          "name": "Abu_Dhabi",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "gainsboro",
        "borderRadius": 4,
        "borderWidth": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <Connect(AddCityModal)
    toggleVisibility={[Function]}
    visible={false}
  />
</View>
`;

exports[`Dashboard screen component should render correctly 2`] = `
<View
  style={
    Object {
      "backgroundColor": "ghostwhite",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "country": "united_arab_Emirates",
          "displayName": "Abu Dhabi",
          "id": "1",
          "name": "Abu_Dhabi",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "gainsboro",
        "borderRadius": 4,
        "borderWidth": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <Connect(AddCityModal)
    toggleVisibility={[Function]}
    visible={true}
  />
</View>
`;
